File: tfidf.py
from sklearn.feature_extraction.text import TfidfVectorizer

documents = [
    "This is the first document.",
    "This document is the second document.",
    "And this is the third one.",
    "Is this the first document?"
]

vectorizer = TfidfVectorizer()
X = vectorizer.fit_transform(documents)

print(vectorizer.get_feature_names())
print(X.toarray())

File: tfidf.rs
use ndarray::Array2;
use rust_tfidf::TfidfVectorizer;

fn main() {
    let documents = vec![
        "This is the first document.",
        "This document is the second document.",
        "And this is the third one.",
        "Is this the first document?",
    ];

    let vectorizer = TfidfVectorizer::new();
    let matrix: Array2<f64> = vectorizer.fit_transform(&documents);

    println!("{:?}", matrix);
}
